packages/blade/src/components/Input/TextArea/__tests__/TestArea.web.test.tsx

```diff
@@ -0,0 +1,264 @@
+    const onChange = jest.fn();
+    const user = userEvent.setup();
+    const userName = 'Divyanshu';
+
+    const { getByLabelText } = renderWithTheme(
+      <TextArea label={label} name="name" onChange={onChange} />,
+    );
+
+    const input = getByLabelText(label);
+    await user.type(input, userName);
+
+    // should be called for each keystroke
+    expect(onChange).toHaveBeenCalledTimes(userName.length);
+    expect(onChange).toHaveBeenLastCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should handle onBlur', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const userName = 'Divyanshu';
+    const onBlur = jest.fn();
+
+    renderWithTheme(
+      <TextArea
+        label={label}
+        name="name"
+        defaultValue={userName}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
+        onBlur={onBlur}
+      />,
+    );
+
+    // shifts user focus and therefore blurs the focussed input
+    await user.tab();
+    expect(onBlur).toHaveBeenCalledTimes(1);
+    expect(onBlur).toHaveBeenCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should be focussable', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const userName = 'Divyanshu';
+
+    const { getByLabelText } = renderWithTheme(
+      <TextArea label={label} name="name" defaultValue={userName} />,
+    );
+
+    const input = getByLabelText(label);
+
+    expect(input).not.toHaveFocus();
+    await user.tab();
+    expect(input).toHaveFocus();
+  });
+
+  it('should set value as an uncontrolled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const { getByLabelText } = renderWithTheme(
+      <TextArea label={label} defaultValue={valueInitial} />,
+    );
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, ' Maithani');
+    expect(input).toHaveValue(valueFinal);
+  });
+
+  it('should set value as a controlled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const valueInitial = 'Divyanshu';
+    const valueFinal = 'Divyanshu Maithani';
+
+    const ControlledInputExample = (): ReactElement => {
+      const [value, setValue] = React.useState<string | undefined>(valueInitial);
+
+      return (
+        <>
+          <TextArea label={label} value={value} onChange={({ value }) => setValue(value)} />
+          <p data-testid="input-value">{value}</p>
+        </>
+      );
+    };
+
+    const { getByLabelText, getByTestId } = renderWithTheme(<ControlledInputExample />);
+
+    const input = getByLabelText(label);
+    expect(input).toHaveValue(valueInitial);
+
+    await user.type(input, ' Maithani');
+    expect(getByTestId('input-value')).toHaveTextContent(valueFinal);
```

Comment:
why do we need to check the value of paragraph? we shall assert it against the actual textarea right?

Reply:
Changed.